src/coreclr/src/jit/lclvars.cpp

```diff
@@ -5454,12 +5484,6 @@ int Compiler::lvaAssignVirtualFrameOffsetToArg(unsigned lclNum,
             offset += fieldVarDsc->lvFldOffset;
         }
     }
-    // For an independent promoted struct field we also assign the parent struct stack offset
-    else if (varDsc->lvIsStructField)
-    {
-        noway_assert(varDsc->lvParentLcl < lvaCount);
-        lvaTable[varDsc->lvParentLcl].lvStkOffs = varDsc->lvStkOffs;
```

Comment:
Do we leave it set to `BAD_STK_OFFS` after this change?

Reply:
No, this was duplicative; the field offsets are set in the promoted struct case, and don't need to be set again when they are encountered.